src/lib/staged_ledger_hash/staged_ledger_hash.ml

```diff
@@ -196,64 +153,63 @@ module Non_snark = struct
         Lazy.force dummy )
 end
 
-module Stable = struct
-  module Poly = struct
-    module Stable = struct
-      module V1 = struct
-        module T = struct
-          type ('non_snark, 'pending_coinbase_hash) t =
-            { non_snark: 'non_snark
-            ; pending_coinbase_hash: 'pending_coinbase_hash }
-          [@@deriving bin_io, sexp, eq, compare, hash, yojson, version]
-        end
-
-        include T
-      end
-
-      module Latest = V1
+module Poly = struct
+  [%%versioned
+  module Stable = struct
+    module V1 = struct
+      type ('non_snark, 'pending_coinbase_hash) t =
+        {non_snark: 'non_snark; pending_coinbase_hash: 'pending_coinbase_hash}
+      [@@deriving sexp, eq, compare, hash, yojson]
     end
+  end]
 
-    type ('non_snark, 'pending_coinbase_hash) t =
-          ('non_snark, 'pending_coinbase_hash) Stable.Latest.t =
-      {non_snark: 'non_snark; pending_coinbase_hash: 'pending_coinbase_hash}
-    [@@deriving sexp, compare, hash, yojson]
-  end
+  type ('non_snark, 'pending_coinbase_hash) t =
+        ('non_snark, 'pending_coinbase_hash) Stable.Latest.t =
+    {non_snark: 'non_snark; pending_coinbase_hash: 'pending_coinbase_hash}
+  [@@deriving sexp, compare, hash, yojson]
+end
 
+[%%versioned
+module Stable = struct
   module V1 = struct
-    module T = struct
-      (** Staged ledger hash has two parts
+    (** Staged ledger hash has two parts
       1) merkle root of the pending coinbases
       2) ledger hash, aux hash, and the FIFO order of the coinbase stacks(Non snark).
       Only part 1 is required for blockchain snark computation and therefore the remaining fields of the staged ledger are grouped together as "Non_snark"
       *)
-      type t =
-        ( Non_snark.Stable.V1.t
-        , Pending_coinbase.Hash.Stable.V1.t )
-        Poly.Stable.V1.t
-      [@@deriving bin_io, sexp, eq, compare, hash, yojson, version]
-    end
+    type t =
+      ( Non_snark.Stable.V1.t
+      , Pending_coinbase.Hash.Stable.V1.t )
+      Poly.Stable.V1.t
+    [@@deriving sexp, eq, compare, hash, yojson]
 
-    include T
-    include Registration.Make_latest_version (T)
-    include Hashable.Make_binable (T)
-  end
+    let to_latest = Fn.id
 
-  module Latest = V1
-
-  module Module_decl = struct
-    let name = "staged_ledger_hash"
+    module T = struct
+      type typ = t [@@deriving sexp, compare, hash]
+
+      type t = typ [@@deriving sexp, compare, hash]
+
+      (* can't use derive bin_io which will trigger the linter *)
+      [%%define_existing
```

Comment:
Not a fan of this workaround; I'll tweak the versioning ppx to accept `module T`s when there's a later include.

Reply:
OK ... I think `define_existing` may still be useful elsewhere. Maybe a better name is `define_from_scope`.